src/dynamic_batch_scheduler.cc

```diff
@@ -359,18 +369,7 @@ DynamicBatchScheduler::BatcherThread(const int nice)
           continue;
         }
 
-        {
-          // The wait_for_slots conditional can be blocking till the slots
-          // are available for execution. Need to explicitly release the
-          // outer lock to allow Enqueue threads above to make progress.
-          lock.unlock();
-          // Use slot lock to wait for the slot availability.
-          std::mutex slot_mu;
-          std::unique_lock<std::mutex> slot_lock(slot_mu);
-          cv_.wait(slot_lock, wait_for_slots);
-          // Recapture the outer most lock to keep making progress.
-          lock.lock();
-        }
+        WaitForPayloadSlotAvailable(&lock, default_wait_microseconds);
```

Comment:
default_wait_microseconds is a fixed value, wouldn't there still be delay of 0.5 seconds? Or that is acceptable for rejecting requests?

Reply:
I think 0.5s should be fine for now. What they were complaining about is timeout of 2s in the scenario, when a model executes for 10s, so timeout requests are waiting for 8 extra seconds to be returned.

We can always adjust it upon their feedback

Reply:
Yes. I think we can have them try this version first, we can always reduce the delay interval if they ask. I have filed an enhancement ticket to remove the need of a manually set interval.

Reply:
> wouldn't there still be delay of 0.5 seconds? Or that is acceptable for rejecting requests?

I think it is acceptable, since they mentioned "[some large model] could last for tens of seconds" and it becomes a concern, which 0.5 seconds is significantly less than tens of seconds.